src/memory.jl

```diff
@@ -176,6 +176,26 @@ function free(buf::HostBuffer)
     end
 end
 
+##
+# Pinned memory management
+#
+# - IdDict does not free the memory
+# - WeakRef dict does not unique the key by objectid
+# TODO:
+# - Deal with memory regions/views
+const __pinned_memory = Dict{Ptr, WeakRef}()
+
+function pin(a, flags=0)
```

Comment:
```suggestion
function pin(a::Array, flags=0)
```

since we don't want to deal with views.